src/plugins/discover/public/embeddable/get_search_embeddable_factory.tsx

```diff
@@ -181,6 +159,35 @@ export const getSearchEmbeddableFactory = ({
         }
       );
 
+      /**
+       * If the parent is providing context, then the embeddable state transfer service can be used
+       * and editing should be allowed; otherwise, do not provide editing capabilities
+       */
+      const parentApiContext =
+        apiHasParentApi(api) && apiHasAppContext(api.parentApi)
+          ? api.parentApi.getAppContext()
+          : undefined;
+      if (parentApiContext) {
+        api.isEditingEnabled = startServices.isEditable;
```

Comment:
I wanted to do this instead:

```suggestion
      if (apiHasParentApi(api) && apiHasAppContext(api.parentApi)) {
        const parentApiContext = api.parentApi.getAppContext();
        api.isEditingEnabled = startServices.isEditable;
```

The problem is, we have a fallback `getAppContext` in the generic embeddable that will return `undefined` if we comment out the Dashboard's `getAppContext` -  so this is more type safe. Perhaps `getAppContext` should return `EmbedddableAppContext | undefined` to make this clear...

Reply:
Made `getAppContext` conditionally return `undefined` in d0eaa4808be42bb53f2027e6cba740a40f5940e0